unison-core/src/Unison/Name.hs

```diff
@@ -134,9 +144,33 @@ suffixes (Name' names) =
   map (Name' . NonEmpty.fromList) (NonEmpty.init (NonEmpty.tails names))
 
 makeAbsolute :: Name -> Name
-makeAbsolute n | toText n == "."                = Name ".."
-               | Text.isPrefixOf "." (toText n) = n
-               | otherwise                      = Name ("." <> toText n)
+makeAbsolute n | toText n == "." = Name ".."
+               | isAbsolute n    = n
+               | otherwise       = Name ("." <> toText n)
+
+countDots :: Name -> Int
+countDots = Text.count "." . Text.dropEnd 1 . toText
+
+segments :: Name -> [NameSegment]
+segments (Name name) = fmap NameSegment (Text.splitOn "." name)
+
+isLower :: Name -> Bool
+isLower = Text.all Char.isLower . Text.take 1 . toText
+
+isAbsolute :: Name -> Bool
+isAbsolute (Name name) = Text.isPrefixOf "." name && name /= "."
+
+splits :: Name -> [([NameSegment], Name)]
+splits =
+  map (map NameSegment *** unsafeFromText) . oldSplits
+
+-- > oldSplits "x" == [([], "x")]
+-- > oldSplits "A.x" == [(["A"], "x")]
+-- > oldSplits "A.B.x" == [(["A"], "B.x"), (["A.B"], "x")]
+oldSplits :: Name -> [([Text], Text)]
+oldSplits (Name n) = let ns = Text.splitOn "." n
+                     in dropEnd 1 (inits ns `zip` (map dotConcat $ tails ns))
+  where dotConcat = Text.concat . intersperse "."
```

Comment:
I wonder where these are used.

Reply:
`oldSplits` is used in the term printer. I initially moved it into `Name` and called it `splits`, which (IMO) has a more correct type, but left `oldSplits` that just returns `([Text], Text)` so I didn't have to refactor that term printer code. I think `oldSplits` should be replaced by `splits` eventually.